src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx

```diff
@@ -13,78 +12,94 @@ import { ApprovalState } from 'hooks/useApproveCallback'
+enum ErrorMsgs {
+  Balance = 'Insufficient balance to invest. Adjust the amount or go back to remove this investment option',
+  Input = 'Input amount is bigger then available investment amount',
 }
 
 export default function InvestOption({ approveData, claim, optionIndex }: InvestOptionProps) {
   const { currencyAmount, price, cost: maxCost } = claim
   const { updateInvestAmount } = useClaimDispatchers()
-  const { investFlowData } = useClaimState()
+  const { investFlowData, activeClaimAccount } = useClaimState()
 
-  const investedAmount = useMemo(() => investFlowData[optionIndex].investedAmount, [investFlowData, optionIndex])
+  const { account } = useActiveWeb3React()
 
   const [percentage, setPercentage] = useState<string>(INVESTMENT_STEPS[0])
+  const [typedValue, setTypedValue] = useState<string>('0')
+  const [inputError, setInputError] = useState<string>('')
 
-  const { account } = useActiveWeb3React()
+  const investedAmount = investFlowData[optionIndex].investedAmount
 
   const token = currencyAmount?.currency
-
   const balance = useCurrencyBalance(account || undefined, token)
 
   const decimals = balance?.currency?.decimals
 
-  const handleStepChange = useCallback(
+  const isSelfClaiming = account === activeClaimAccount
+
+  // on invest max amount click handler
+  const setMaxAmount = useCallback(() => {
+    if (!maxCost || !balance) {
+      return
+    }
+
+    const value = maxCost.greaterThan(balance) ? balance : maxCost
+    const amount = value.quotient.toString()
+
+    updateInvestAmount({ index: optionIndex, amount })
+    setTypedValue(formatSmart(value, decimals) || '')
+    setPercentage(INVESTMENT_STEPS[INVESTMENT_STEPS.length - 1])
+  }, [balance, decimals, maxCost, optionIndex, updateInvestAmount])
+
+  // on input field change handler
+  const onInputChange = useCallback(
     (value: string) => {
-      if (!maxCost || !balance) {
+      setTypedValue(value)
+      setInputError('')
+
+      const input = parseFloat(value)
+
+      if (!value.length || isNaN(input)) {
+        updateInvestAmount({ index: optionIndex, amount: '0' })
+        setPercentage('0')
         return
       }
 
-      const scaledCurrencyAmount = _scaleValue(maxCost, value)
+      if (!value || !maxCost || !balance || !token) {
+        return
+      }
 
-      updateInvestAmount({ index: optionIndex, amount: scaledCurrencyAmount.quotient.toString() })
-      setPercentage(value)
-    },
-    [balance, maxCost, optionIndex, updateInvestAmount]
-  )
+      // parse input to BitInt string
+      const parsed = CurrencyAmount.fromRawAmount(token, parseUnits(input.toString(), decimals).toString())
 
-  const onMaxClick = useCallback(() => {
-    if (!maxCost || !balance) {
-      return
-    }
+      // calculate percent
+      const maxValue = maxCost.greaterThan(balance) ? balance : maxCost
+      const hundred = parseUnits('100', decimals).toString()
+      const percent = parsed.divide(maxValue).multiply(hundred)
```

Comment:
This is a bit confusing. Why multiply 100 * decimals?
I would expect the percentage to be between 0 and 1, or (since you multiply it by 100) between 0 and 100.

If `parsed` and `maxValue` are in the same units (atoms?) the result should be in the same unit, no?
1 * 10**18 / 100 * 10**18 => 0.01

Reply:
@alfetopito i made changes while i was reviewing and pushed so this is outdated